src/session/error.rs

```diff
@@ -19,6 +19,12 @@ pub struct Error {
 pub enum ErrorKind {
     /// The action execution failed.
     ExecutionFailure,
+    /// I/O error.
+    #[cfg(target_family = "unix")]
+    IoError(std::io::ErrorKind),
+    /// Windows error.
+    #[cfg(target_os = "windows")]
+    WindowsError,
```

Comment:
For me these are subcases of `ExecutionFailure` and thus should be defined as errors for individual actions. Take a look how the `listdir` action [does it][1].

[1]: https://github.com/google/rrg/blob/a7eb4a0a2f9c3818eef760e96d562100868d5fc5/src/action/listdir.rs#L36-L72

Reply:
I created a new error type. The signature of the `get_hostname` function now throws `std::io::Error` for all platforms